assets/js/app.js

```diff
@@ -63,4 +63,42 @@ var app = angular.module("Lucere", ["ngResource", "ngRoute", "dndLists"])
         });    
       }
     });
-  }]);
\ No newline at end of file
+  }])
+  .run(["$rootScope", "$location", "AuthService", "$route", "$routeParams", function($rootScope, $location, AuthService, $route, $routeParams) {
+    $rootScope.$on("$routeChangeSuccess", function(event, next, current) {
+      var routeParams = $route.current.params;
+      var libId = routeParams.libraryId;
+
+      var isAdminRoute = /(admin\/)/.test($location.path());
+
+      // If on a library route, check to see if user should have access
+      if(libId) {
+        AuthService.currentUser(function(user) {
+          if(user && user.teams) {
+            var allowAccess = false;
+
+            // Users accessing admin route must be admins
+            if(isAdminRoute) {
+              user.teams.forEach(function(team) {
+                user.administrating.forEach(function(admining) {
```

Comment:
Is forEach a synchronous function? If it isn't, we cannot guarantee that allowAccess has the proper value at the last if statement